PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -69,16 +75,9 @@ public function __construct(
         stream_context_set_option($this->context, 'socket', 'tcp_nodelay', true);
 
         $options = stream_context_get_options($this->context);
-        if (!empty($options['ssl'])) {
-            if (isset($ssl_protocol)) {
-                $this->protocol = $ssl_protocol;
-            } else {
-                // https://www.php.net/manual/en/transports.inet.php
-                if (PHP_VERSION_ID >= 70200) {
-                    $this->protocol = 'tls';
-                } else {
-                    $this->protocol = 'ssl';
-                }
+        if (!empty($options['ssl']) && !isset($options['ssl']['crypto_method']) && $ssl_protocol !== 'tcp') {
```

Comment:
This use did not make sense, because $ssl_protocol value is not passed in here, so I have now removed it.

The point was that if $ssl_protocol is 'tcp', then we should not enable encryption. But there is no point in passing it with non-empty $options['ssl'].